index.js

```diff
@@ -54,11 +55,23 @@ function decorateLayer(layer, keysToKeep, newProps) {
         keepLookup[keys.indexOf(keysToKeep[i])] = true;
     }
 
+    if (requiredKeys && requiredKeys.length) {
+        for (i = 0; i < requiredKeys.length; i++) {
+            var index = keys.indexOf(requiredKeys[i]);
+            if (index !== -1) requiredIndices.push(index);
+        }
+    }
```

Comment:
`indexOf` when doing `hasAllKeys` check can be expensive if we have many keys. Can we do it in the same manner as the `keepLookup` logic above — building a hashmap of required indices, and then checking against it instantly?